### DISCUSSION TITLE
Valid values for the ifExists query parameter

### DISCUSSION THREAD

**Discussion by abclabs-jon** [Apicurio/apicurio-registry#5454] [Q&A] [answered]
In the [API documentation](https://www.apicur.io/registry/docs/apicurio-registry/3.0.x/assets-attachments/registry-rest-api.htm#tag/Artifacts/operation/createArtifact) is says that the query parameter ifExists can have the values:

- FAIL (default)
- UPDATE
- RETURN
- RETURN_OR_UPDATE

This is not the case when running the docker image though (apicurio/apicurio-registry:3.0.3), the only one of the above that works is FAIL. If I use one of the others, I get this:

`{
  "detail": "IllegalArgumentException: No enum constant io.apicurio.registry.rest.v3.beans.IfArtifactExists.UPDATE",
  "title": "RESTEASY003870: Unable to extract parameter from http request: jakarta.ws.rs.QueryParam(\"ifExists\") value is 'UPDATE'",
  "status": 404,
  "name": "NotFoundException"
}`

Looking in the code, it seems the values that actually work are FAIL, CREATE_VERSION and FIND_OR_CREATE_VERSION.

Is the API documentation wrong or am doing something wrong?

**Comment by EricWittmann** [comment, marked as answer]
You are correct. The text description in the API documentation is old. Those values have been updated, and the description of that endpoint was not updated to reflect that. I will get that fixed up. Thanks for letting me know.
